api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -301,6 +301,12 @@
      */
     int getId();
 
+    /**
+     * Return the number of windows before the next checkpoint including the current window.
+     * @return Number of windows from checkpoint, 1 if the checkpoint will be after the current window
+     */
+    int getWindowsFromCheckpoint();
```

Comment:
@gauravgopi123 I had debated 0 or 1 myself. The checkpoint is not in current window but immediately after the end of the window so for the user I thought it would be more intuitive to go with 1